rockcraft/parts.py

```diff
@@ -191,21 +190,4 @@ def validate_part(data: Dict[str, Any]) -> None:
 
     :param data: The part data to validate.
     """
-    if not isinstance(data, dict):
-        raise TypeError("value must be a dictionary")
-
-    # copy the original data, we'll modify it
-    spec = data.copy()
-
-    plugin_name = spec.get("plugin")
-    if not plugin_name:
-        raise ValueError("'plugin' not defined")
-
-    plugin_class = plugins.get_plugin_class(plugin_name)
-
-    # validate plugin properties
-    plugin_class.properties_class.unmarshal(spec)
-
-    # validate common part properties
-    part_spec = plugins.extract_part_properties(spec, plugin_name=plugin_name)
-    PartSpec(**part_spec)
+    craft_parts.validate_part(data)
```

Comment:
are you doing it this way to provide some form of abstraction?

Reply:
I'm accessing craft-parts only from the parts module. This tries to mirror snapcraft which accesses craft-parts only in the parts package.